lib/TokenManager.js

```diff
@@ -99,6 +99,27 @@ function get(storage, key) {
   return tokenStorage[key];
 }
 
+function getAsync(sdk, tokenMgmtRef, storage, key) {
+  var token = get(storage, key);
+  var deferred = Q.defer();
+
+  if (token && (token.expiresAt * 1000 <= Date.now())) {
```

Comment:
This evaluation is also duplicated in `setRefreshTimeout()`, might be nice to make this into a function.

Reply:
We're actually comparing the time here, expecting a `boolean` response here as opposed to calculating the required `refreshWait` time in `setRefreshTimeout`.

We could pull the `refreshWait` calculation into a function, but we're still going to need to evaluate it:

```javascript
function calculateRefreshTime(expiration) {
  return expiration * 1000 - Date.now();
}

...

// This seems less clear to me what is going on
if (token && (calculateRefreshTime(token.expiresAt) <= 0)) {
    if (tokenMgmtRef.autoRefresh) {
      ...
    }
   ...
}

```